test/test-fs.c

```diff
@@ -687,24 +689,48 @@ TEST_IMPL(fs_file_loop) {
   return 0;
 }
 
-static void check_utime(const char* path, double atime, double mtime) {
+
+static void check_utime_ex(const char* path,
+                           double btime,
+                           double atime,
+                           double mtime) {
   uv_stat_t* s;
   uv_fs_t req;
   int r;
 
   r = uv_fs_stat(loop, &req, path, NULL);
   ASSERT(r == 0);
-
   ASSERT(req.result == 0);
   s = &req.statbuf;
 
+#if defined(__APPLE__) || defined(_WIN32)
+  /* When check_utime_ex is called with a btime of NAN, this means that we are
+   * checking uv_fs_utime and uv_fs_futime results, which SHOULD NOT allow the
+   * caller to alter the btime.  Well some utime implementations, like FreeBSD,
+   * have conditions where btime can be altered via utime even though btime is
+   * not an argument.  The conditions to identify this are impossible to check
+   * at test time so we will not check that btime is unaltered when checking
+   * the results of uv_fs_utime and uv_fs_futime.
+   */
+  if (!isnan(btime)) {
+    /* Make sure the birth/creation time was altered as expected. */
+    ASSERT(s->st_birthtim.tv_sec + (s->st_birthtim.tv_nsec / 1000000000.0)
+      == btime);
```

Comment:
Style nit: `==` should go on the previous line.

Reply:
I saw this style in other places and followed, same as above.  I can fix it but I'm pretty sure I followed existing convention.

Reply:
I went ahead and address this nit, I guess I chose the wrong example to follow.